### PR TITLE
Remove sources with LEGACY STATE from registry

### PR DESCRIPTION
**Pull request by evantahler**
[Slack Thread](https://airbytehq-team.slack.com/archives/C03AS1GAQV6/p1707324981920019)

[From this Mb Query](https://airbyte.metabaseapp.com/question/8681-cloud-source-connectors-active-in-the-last-30-days) # of Connections active in the last 30 days:
source-talkdesk-explore - 0
source-kustomer-singer - 0
source-courier - 0
source-zuora - 1
source-dv-360 - 0
source-linnworks - 0
source-search-metrics - 0
source-appstore-singer - 0
source-recurly - 7

This PR also coincides with hiding these connectors from Cloud by adding the actors to the `connectors.hideActorDefinitionFromList` Feature Flag

Screenshot 2024-02-08 at 10 57 58 AM

### CODE REVIEW

File: airbyte-integrations/connectors/source-kustomer-singer/metadata.yaml [hunk 1]
```diff
@@ -13,11 +13,11 @@ data:
...
       # TODO: Set enabled=true after `airbyte-lib-validate-source` is passing.
       packageName: airbyte-source-kustomer-singer
-  registries:
+  registries: # Removed from registries due to LEGACY STATE
     cloud:
-      enabled: true
+      enabled: false
     oss:
-      enabled: true
+      enabled: false
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-kustomer-singer/metadata.yaml:20]: Note that this PR also removes these connectors from the OSS repo

File: airbyte-integrations/connectors/source-linnworks/metadata.yaml
Comment (alafanechere) [inline airbyte-integrations/connectors/source-linnworks/metadata.yaml:1]: 😢 I supported the merge of this contributed connectors two years ago

File: airbyte-integrations/connectors/source-appstore-singer/metadata.yaml
Comment (alafanechere) [inline airbyte-integrations/connectors/source-appstore-singer/metadata.yaml:1]: I think its also safe to remove other singer based connector if any. They're all probably using the LEGACY STATE. And we can remove the generator for singer.

**Review by alafanechere — APPROVED** [review]

File: airbyte-integrations/connectors/source-appstore-singer/metadata.yaml
Comment (evantahler) [reply airbyte-integrations/connectors/source-appstore-singer/metadata.yaml:1]: I think there are only 2 left!

**Comment by octavia-approvington** [comment]
Check that box <br/> TODO complete

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**